### PR TITLE
Implementation of the nf-core YaHS module

### PR DESCRIPTION
**Pull request by ksenia-krasheninnikova**
## PR checklist

Closes [#2569](https://github.com/nf-core/modules/issues/2569) 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/yahs/main.nf
Comment (pinin4fjords) [inline modules/nf-core/yahs/main.nf:13]: I think, according to [current guidelines](https://nf-co.re/docs/contributing/modules#general), that these should all go through `ext.args`.

File: modules/nf-core/yahs/main.nf
Comment (pinin4fjords) [inline modules/nf-core/yahs/main.nf:28]: As referenced above, I think all the options here can come directly through from ext.args without the need for explicit `val` inputs.

File: modules/nf-core/yahs/main.nf
Comment (pinin4fjords) [inline modules/nf-core/yahs/main.nf:37]: You shouldn't redirect STDOUT, Nextflow will use that to populate .command.out.

File: modules/nf-core/yahs/meta.yml
Comment (pinin4fjords) [inline modules/nf-core/yahs/meta.yml:38]: Unless my points on `val` inputs are contradicted by someone else, the related documentation will also need to be removed (we defer to the upstream documentation).

File: modules/nf-core/yahs/main.nf
Comment (pinin4fjords) [inline modules/nf-core/yahs/main.nf:18]: Could you vertically align the `emit:`s for readability? Have a look at other modules to see what I mean.

File: modules/nf-core/yahs/meta.yml
Comment (pinin4fjords) [inline modules/nf-core/yahs/meta.yml:28]: This should be `ref`, no?

File: tests/config/test_data.config
Comment (pinin4fjords) [inline tests/config/test_data.config:553]: You should use existing data from the [nf-core test data repository](https://nf-co.re/docs/contributing/test_data_guidelines), or else PR some new (small) test files if there's nothing suitable.

These will be used in CI etc, and we don't want to be hammering the Sanger servers repeatedly.

**Review by pinin4fjords — CHANGES_REQUESTED** [review]

**Review by nvnieuwk — CHANGES_REQUESTED** [review]
Good job! Here are some suggestions :)

File: modules/nf-core/yahs/main.nf
Comment (pinin4fjords) [inline modules/nf-core/yahs/main.nf:19]: re-align ;-)

File: modules/nf-core/yahs/main.nf
Comment (pinin4fjords) [inline modules/nf-core/yahs/main.nf:25]: @nvnieuwk meant that you needed to pass task.ext.prefix like [this](https://github.com/nf-core/modules/blob/ee80d14721e76e2e079103b8dcd5d57129e584ba/modules/nf-core/gsea/gsea/main.nf#L45) and [this](https://github.com/nf-core/modules/blob/ee80d14721e76e2e079103b8dcd5d57129e584ba/tests/modules/nf-core/gsea/gsea/nextflow.config#L15).

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
Almost there!

File: modules/nf-core/yahs/main.nf
Comment (nvnieuwk) [reply modules/nf-core/yahs/main.nf:25]: Yes, my bad for not being clear enough about it :D

Comment (pinin4fjords) [reply modules/nf-core/yahs/main.nf:25]: @ksenia-krasheninnikova that might mean passing the prefix to the tool to determine output file names, or else renaming the files afterwards (as I had to do in the GSEA case).

Comment (ksenia-krasheninnikova) [reply modules/nf-core/yahs/main.nf:25]: Thank you both! I hope the last update does it.

**Review by nvnieuwk — APPROVED** [review]
LGTM! if tests pass

File: modules/nf-core/yahs/meta.yml
Comment (nvnieuwk) [inline modules/nf-core/yahs/meta.yml:59]: Can you also update the pattern here?

File: modules/nf-core/yahs/main.nf
Comment (nvnieuwk) [inline modules/nf-core/yahs/main.nf:18]: The output files aren't being detected, are you sure this can find them all?

Comment (pinin4fjords) [reply modules/nf-core/yahs/main.nf:18]: I think the test.yml just needs updating for the new prefix ('test')

File: modules/nf-core/yahs/meta.yml
Comment (pinin4fjords) [inline modules/nf-core/yahs/meta.yml:54]: No, you were right before- these should hold the wildcards. It's test.yml needs fixing.

**Review by pinin4fjords — APPROVED** [review]
Looking good - thanks for addressing the feedback.

**Comment by ksenia-krasheninnikova** [comment]
Thank you for help @pinin4fjords and @nvnieuwk!

**Pull request merged**
